upscale_droop/upscale_droop_script.R

```diff
@@ -0,0 +1,220 @@
+#           This would mean each class would at be at its maximum capacity output immediately before the event (as normalised value is 1 at pre event interval)
+#           the traces are then scaled by the external capacity factor, 0.X, such that the outputs is at X% its maximum capacity during the pre event interval   
+
+###################################################################################################################
+
+# Option 1
+if(site_norm) {
+  # Get the average site performance factor in underlyng data for each class per timestep
+   site_performance_factor <- group_by(UD, site_id, ts) %>%
+     summarise(site_performance_factor = first(site_performance_factor),
+               manufacturer = first(manufacturer), Standard_Version = first(Standard_Version),
+               StdComplianceCombined = first(StdComplianceCombined))
+  
+   # frst join UD with proportions to identify which OEMs have been assigned to "other" (come up as NA in proportion_cap)
+   site_performance_factor <- left_join(site_performance_factor, Proportions[c("StdComplianceCombined", "proportion_capacity", "manufacturer")],
+                                        by = c("StdComplianceCombined","manufacturer"))
+   
+   # combine any OEM's into 'Other' that are < 30 samples or have unknown, multiple / mixed. This apears as with a proportion capacity of 'na' following the join  
+   site_performance_factor <- combine_erroneous_OEMs(site_performance_factor)
+   
+   
+  # Combine and get average site performance capacity factor for each class with the OEMs > 30 samples. 
+   site_performance_factor <- group_by(site_performance_factor, ts,
+                                       manufacturer, Standard_Version, StdComplianceCombined) %>%
+      summarise(average_site_performance_factor = mean(site_performance_factor))
+  
+   
+   # add back the proportion capacities
+   upscale_MW_profile_OEM <- left_join(site_performance_factor, Proportions[c("StdComplianceCombined", "proportion_capacity", "manufacturer")],
+                                       by = c("StdComplianceCombined","manufacturer"))
+   
+   # upscale per class 
+   upscale_MW_profile_OEM <- mutate(upscale_MW_profile_OEM, upscale_MW = average_site_performance_factor*proportion_capacity/1000)
+   
+   # Combine the total from each OEM
+   upscale_MW_profile <- group_by(upscale_MW_profile_OEM, ts, StdComplianceCombined)
+   upscale_MW_profile <- summarise(upscale_MW_profile, upscale_MW = sum(upscale_MW))
+
+   pivot_upscaled_MW_profile <- pivot_wider(select(upscale_MW_profile, c("ts","StdComplianceCombined", "upscale_MW")), names_from = StdComplianceCombined,
+                                             values_from =upscale_MW)
+   write.csv(pivot_upscaled_MW_profile, paste(output_directory,"droop_compliance_upscale_by_OEM_site_normalisation.csv",sep=""), row.names = FALSE)
```

Comment:
Might be nice to have the filename defined at the top so if you're doing multiple runs it's easier to rename it